### ISSUE TITLE
Add scoping to the library

### ISSUE THREAD

**Issue by michalbe** [ajakubo1/CanvasMenu#14]
Creating separate global constructor for every component will cause mess in the code with more than just two components. So instead of:

``` javascript
var myMenu = new Menu({});
var button = new Button({});
```

Something like any of the solution below should be implemented:

``` javascript
var myMenu = new CM.Menu({});
var button = new CM.Button({});
```

``` javascript
var myMenu = new Menu({});
var button = new Menu.Button({});
```

``` javascript
var myMenu = new Menu({});
// DOM like API, inspired by document.createElement()
var button = myMenu.createButton({});
```

**Comment by michalbe** [comment]
@ajakubo1 which one do you prefer? I can start implementing some of this.

**Comment by ajakubo1** [comment]
I think that the last one would be the best. Implementing it this way would allow us attach the button directly to menu which we've created.

Buttons in essence should not be separate entities from Menus, as they don't support all of the functionalities on their own (e.g. render callbacks).

But this also adds another questions - should we have the `append` function opened in `Menu`? I don't see the point of that, unless you want to duplicate the same button in a couple of `Menu` screens?

**Comment by ajakubo1** [comment]
@michalbe, I've added CM scope to all of the elements. I've had some time which I spent thinking about the scopes, and came to conclusion that the best solution is to mix-up the first and last solutions. I'm doing that because of creation of new types of elements to the library. I want those elements to be dynamically attached to the library, so that users can decide whether they need some specific element or not per deployment.

Having written that, I will implement as well a way of creating the `Button` from `Menu` level, just by typing:

``` javascript
var button = myMenu.createButton({});
```

Which will be automatically attached to `myMenu` object.

So basically, there is a possibility of calling the `CM.Button({})` constructor from the outside. But it won't make a lot of sense doing that in a couple of commits, as it will be handled via `createButton`, `createSwitch`, `createRadio`.

If you (or anyone) come up with an idea which will be flexible as well as it will hide all of the elements except `CM.Menu()`, let me know. I will gladly consider it and rebuild the library.

**Comment by michalbe** [comment]
Hmm. So maybe, if we want to support multiple components, we should change it to something like `create()` with a parameter?
Like:

``` javascript
var button = myMenu.create('button', {});
var switch = myMenu.create('switch', {});
var radio = myMenu.create('radio', {});
```

This solution is closest to the DOM API we can ever get :)

**Comment by michalbe** [comment]
I've created a patch that uses `create('button', {})` instead of `createButton({})`. I believe this solution is more generic and will be simpler when we'll add new components.

**Comment by ajakubo1** [comment]
Great, @michalbe! And I like that it takes string into consideration and then resoves it into Object property, this will allow users to write their own elements and attach them to the library :3.
